otelcollector/scripts/main.sh

```diff
@@ -49,6 +45,23 @@ echo_var "CCP_METRICS_ENABLED" "$CCP_METRICS_ENABLED"
 echo_var "CONTROLLER_TYPE" "$CONTROLLER_TYPE"
 echo_var "CLUSTER" "$CLUSTER"
 
+# wait for configmap sync container to finish initialization
+settingsChangedFile="/etc/config/settings/inotifysettingscreated"
+if [ "${CCP_METRICS_ENABLED}" == "true" ] && [ ! -f $settingsChangedFile ]; then
+  echo "Waiting for ama-metrics-config-sync container to finish initialization..."
+  while true; do
+    event=$(inotifywait -q -e create --format '%f' $(dirname "$settingsChangedFile"))
+    if [[ "$event" == "$(basename "$settingsChangedFile")" ]]; then
+      break
+    fi
+  done
+fi
+
+#Run inotify as a daemon to track changes to the mounted configmap.
+touch /opt/inotifyoutput.txt
+inotifywait /etc/config/settings --daemon --recursive --outfile "/opt/inotifyoutput.txt" --event create,delete --format '%e : %T' --timefmt '+%s'
+
+
```

Comment:
worth getting Kaveesh to eyeball this now in the context of his work to remove shell scripts